lib/galaxy/webapps/galaxy/api/library_datasets.py

```diff
@@ -155,7 +157,7 @@ def _get_current_roles(self, trans, library_dataset):
         :rtype:     dictionary
         :returns:   dict of current roles for all available permission types
         """
-        return self.serialize_dataset_association_roles(library_dataset)
+        return self.hda_manager.serialize_dataset_association_roles(trans, library_dataset)
```

Comment:
I think this should be `ldda_manager` (added in the same PR, see https://github.com/galaxyproject/galaxy/commit/49693814a9702a855d7df166d27c7a6f4f1faba1#diff-55311aee30c3c44eb2dc3159363be503R47 ) instead of `hda_manager`.

Reply:
Exactly what I think I was thinking - it should inherit the same method as the HDA one (https://github.com/galaxyproject/galaxy/commit/49693814a9702a855d7df166d27c7a6f4f1faba1#diff-1c4ba52684d803d42ae20734665ea77cR317). Let me know if that doesn't work?

Reply:
So something like now?